docker-app/qfieldcloud/core/admin.py

```diff
@@ -584,6 +585,73 @@ def queryset(self, request, queryset):
         return queryset.filter(owner__type=value)
 
 
+class ProjectSecretForm(ModelForm):
+    class Meta:
+        model = Secret
+        fields = ("project", "name", "type", "value", "created_by")
+
+    name = fields.CharField(widget=widgets.TextInput)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.instance.pk and self.instance.type == Secret.Type.ENVVAR:
+            self.fields["value"].widget = widgets.TextInput()
+        else:
+            self.fields["value"].widget = widgets.Textarea()
+
+    def get_initial_for_field(self, field, field_name):
+        if self.instance.pk and field_name == "value":
+            return ""
+        return super().get_initial_for_field(field, field_name)
+
+    def clean(self):
+        cleaned_data = super().clean()
+
+        if self.instance.pk:
+            type = self.instance.type
+        else:
+            type = cleaned_data.get("type")
+        if type == Secret.Type.PGSERVICE:
+            # validate the pg_service.conf
+            value = cleaned_data.get("value")
+            if value:
```

Comment:
This has the effect, that without any value you will not get the failing validation error (yet), and just see "This field is required" until you have added a value (which then gets validated in case of `Secret.Type.PGSERVICE`)